### PR TITLE
TASK-45036: Implement task attachment ACL plugin

### PR DESCRIPTION
**Pull request by aelbenney**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -58,6 +58,7 @@
     <mockito.version>2.0.13-beta</mockito.version>
     <h2database.version>1.2.145</h2database.version>
     <org.lombok.version>1.18.2</org.lombok.version>
+    <addon.exo.ecms.version>6.2.x-ux-SNAPSHOT</addon.exo.ecms.version>
```
Comment (hbenali) [inline pom.xml]: Please move this property after line [50](https://github.com/exoplatform/task/pull/565/files/44567b50f7114e426f237dcd1e7a8cf5c064ce7e..68dc470c9bd86192a1084074b981ca2368576f7f#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8L50).

File: services/src/main/java/org/exoplatform/task/plugin/TaskAttachmentACLPlugin.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import org.exoplatform.task.dto.ProjectDto;
+import org.exoplatform.task.dto.TaskDto;
+import org.exoplatform.task.exception.EntityNotFoundException;
+import org.exoplatform.task.service.TaskService;
+
+public class TaskAttachmentACLPlugin extends AttachmentACLPlugin {
+
+  private static final Log    LOG        = ExoLogger.getLogger(TaskAttachmentACLPlugin.class.getName());
+
+  private static final String entityType = "task";
```
Comment (boubaker) [inline services/src/main/java/org/exoplatform/task/plugin/TaskAttachmentACLPlugin.java]: constant name

File: pom.xml [hunk 3]
```diff
@@ -48,6 +48,7 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <org.exoplatform.social.version>6.2.x-ux-SNAPSHOT</org.exoplatform.social.version>
     <addon.exo.gamification.version>2.2.x-ux-SNAPSHOT</addon.exo.gamification.version>
+    <addon.exo.ecms.version>6.2.x-ux-SNAPSHOT</addon.exo.ecms.version>
```
Comment (boubaker) [inline pom.xml:51]:
```suggestion
    <addon.exo.ecms.version>6.2.x-SNAPSHOT</addon.exo.ecms.version>
```

File: services/src/main/java/org/exoplatform/task/plugin/TaskAttachmentACLPlugin.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+  @Override
+  public String getEntityType() {
+    return entityType;
+  }
+
+  @Override
+  public boolean canView(Identity identity, String entityType, String entityId) {
+    if (!entityType.equals(TaskAttachmentACLPlugin.entityType)) {
+      throw new IllegalArgumentException("Entity type must be" + TaskAttachmentACLPlugin.entityType);
```
Comment (boubaker) [inline services/src/main/java/org/exoplatform/task/plugin/TaskAttachmentACLPlugin.java]:
```suggestion
      throw new IllegalArgumentException("Entity type must be" + entityType);
```

File: services/src/main/resources/conf/portal/configuration.xml [hunk 5]
```diff
@@ -111,4 +111,13 @@
     </component-plugin>
   </external-component-plugins>
 
+  <external-component-plugins>
+    <target-component>org.exoplatform.services.attachments.service.AttachmentService</target-component>
+    <component-plugin>
```
Comment (boubaker) [inline services/src/main/resources/conf/portal/configuration.xml]:
```suggestion
    <component-plugin profiles="ecms">
```

**Review by boubaker — APPROVED** [review]

**Pull request merged**